### PR TITLE
Update Extensions Configuration Secrets and Identity sdk from t1 to t2

### PR DESCRIPTION
**Pull request by Tianyun-G**
1.Update Extensions Configuration Secrets and Identity sdk from t1 to t2

2.Update files:

- i. KeyRotation/KeyRotationHelper.cs
- ii. KeyRotationSample.csproj
- iii. KeyVault/KeyVault.cs
- iv. Program.cs

@scottaddie,@AlexGhiondea,@schaabs,@pakrym and @christothes for notification.

### CODE REVIEW

**Comment by v-xuto** [comment]
@scottaddie,@AlexGhiondea,@schaabs,@pakrym, @christothes   Could you please review this PR?

**Comment by ahsonkhan** [comment]
I don't work on the .NET Identity or KeyVault SDKs, so I don't think I am the appropriate person to review this. Maybe it was meant for someone else :)

**Comment by v-xuto** [comment]
@scottaddie,@AlexGhiondea,@schaabs,@pakrym, @christothes Could you please review this PR?

**Comment by v-xuto** [comment]
@scottaddie,@AlexGhiondea,@schaabs,@pakrym, @christothes Could you please review this PR?

File: KeyVault/KeyVault.cs [hunk 1]
```diff
@@ -1,33 +1,26 @@
...
+﻿using Azure.Identity;
+using Azure.Security.KeyVault.Secrets;
 using System;
 
 namespace KeyRotationSample.KeyVault
 {
     public static class KeyVault
     {
-        public static KeyVaultClient GetKeyVaultClient()
+        public static SecretClient GetSecretClient()
```
Comment (christothes) [inline KeyVault/KeyVault.cs]: It's not clear why we need this class - I'd recommend just calling the `SecretClient` ctor directly in Program.cs

File: Program.cs [hunk 2]
```diff
@@ -31,10 +31,10 @@ public static IHostBuilder CreateHostBuilder(string[] args)
                   
                       if (!string.IsNullOrEmpty(vaultUrl))
                       {
-                         keyVaultClient = KeyVault.KeyVault.GetKeyVaultClient();
+                          secretClient = KeyVault.KeyVault.GetSecretClient();
```
Comment (christothes) [inline Program.cs]: call SecretClient ctor directly here.

File: KeyRotationSample.csproj [hunk 3]
```diff
@@ -9,12 +9,11 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="Azure.Extensions.AspNetCore.Configuration.Secrets" Version="1.2.1" />
+    <PackageReference Include="Azure.Identity" Version="1.4.0" />
```
Comment (christothes) [inline KeyRotationSample.csproj]: 1.5.0 is the latest version.

Comment (Tianyun-G) [reply KeyRotationSample.csproj]: @christothes - I have updated , please review.

File: Program.cs (hunk 2 above)
Comment (Tianyun-G) [reply Program.cs]: @christothes - I have updated , please review.

**Review by christothes — APPROVED** [review]

**Comment by christothes** [comment]
FYI @heaths

**Comment by pakrym** [comment]
LGTM

File: Program.cs [hunk 4]
```diff
@@ -31,10 +31,10 @@ public static IHostBuilder CreateHostBuilder(string[] args)
                   
                       if (!string.IsNullOrEmpty(vaultUrl))
                       {
-                         keyVaultClient = KeyVault.KeyVault.GetKeyVaultClient();
-                         
+                          secretClient = new SecretClient(new Uri(vaultUrl), new DefaultAzureCredential());
+
                          // Add azurekey vault configurations to configuration store. 
-                         config.AddAzureKeyVault(vaultUrl, keyVaultClient, new DefaultKeyVaultSecretManager());
+                         config.AddAzureKeyVault(new Uri(vaultUrl), new DefaultAzureCredential());
```
Comment (heaths) [inline Program.cs]: There should be an overload that lets you pass in the `SecretClient` so another one isn't created here. Even if you do want to create a new one, I recommend sharing the DefaultAzureCredential to save a couple hops.

**Review by heaths — APPROVED** [review]

File: Program.cs (hunk 4 above)
Comment (Tianyun-G) [reply Program.cs]: @heaths - I have updated , please review.

**Review by heaths — APPROVED** [review]

**Pull request merged**